src/DataTable/index.jsx

```diff
@@ -77,6 +78,26 @@ function DataTable({
               selectedRowIds: {},
             };
           }
+          /**
+           Note: We override the `toggleAllRowsSelected` action from react-table
+           because we need to reassign the `selectedRowIds` property depending on
+           the size of the maximum rows selected.
```

Comment:
Why could it be a usability issue? if the user wants to chose the previews selected row may check it again and uncheck the checkboxes that he doesn't want to select, even doing is the last checkbox that will be changed it will be just that one. I think that disable the other rows can be aggressive for the user, I mean can think that is not possible to check them after the selection

Reply:
@johnvente. I spoke with the designer responsible for the existing pattern for `DataTable` to limit the number of maximum selected rows. _We both feel it's preferable to prevent going beyond the maximum configured by disabling selections once the user has reached the maximum._

The rationale for this decision is due to the usability issue of the proposed approach _assuming_ the user will choose the de-select the previously selected row, which is not an assumption we can make. If that previously selected row was not a row the user would have de-selected on their own, they would now need to re-select the row that was programmatically de-selected. We feel this is not efficient for the user and likely to cause confusion; the user should be able to make an intentional choice of which row to deselect.

Further, while `DataTable` should enforce disabling of selection checkboxes, we also believe there is a separate design task to think through how to best communicate the maximum number of rows the user can select as an official part of the `DataTable` UI. For example, in our current use case, we have some micro-copy nearby the `DataTable` to explain to the user that "up to 12 rows may be selected." The design task would help understand whether such messaging be a generic, consistent pattern for all use cases.

Given the follow-up design task to explore this pattern, I recommend the following next steps:

* Implement `maxSelectedRows` prop to disable selection checkboxes once a configured maximum has been reached. For the "Select all" checkbox in the table header, I feel it would be preferable to simply not include a checkbox in the table header rather than having its checkbox disabled.
  * Perhaps we could conditionally hide the "Select all" checkbox when `maxSelectedRows=true` and the number of selected rows <= number of items in the table (i.e., `itemCount`)? 🤔 
* [consider] Understand whether consumers of the `DataTable` component have the ability to get at the current number of selected rows outside of `DataTable` children (i.e., external to the `DataTableContext`).
  * If not, perhaps introducing a callback function prop (e.g., `onRowSelected` or `onMaxSelectedRows`) which would get called by `DataTable` when the user selects a row, or reaches the configured maximum number of rows. Using such props could allow consumers of `DataTable` to render custom messaging where appropriate if/when the user reaches the maximum number of selected rows.

Reply:
Hey @adamstankiewicz Thanks a lot for being pending and giving me that feedback. I will check this ASAP